docs/server-config.md

```diff
@@ -4,47 +4,149 @@
+the defaults.
+
+TabPy parameters explained below and for the logger documentation
+can be found at [`logging.config` documentation page](https://docs.python.org/3.6/library/logging.config.html).
+
+`[TabPy]` parameters:
+
+- `TABPY_PORT` - port for TabPy to listen on. Default value - `9004`.
+- `TABPY_QUERY_OBJECT_PATH` - query objects location. Used with models, see
+  [TabPy Tools documentation](tabpy-tools.md) for details. Default value -
+  `/tmp/query_objects`.
+- `TABPY_STATE_PATH` - state location for Tornado web server. Default
+   value - `tabpy/tabpy_server` subfolder in TabPy package folder.
+- `TABPY_STATIC_PATH` - location of static files (index.html page) for
+  TabPy instance. Default value - `tabpy/tabpy_server/static` subfolder in
+  TabPy package folder.
+- `TABPY_PWD_FILE` - path to password file. Setting up this parameter
+  makes TabPy require credentials with HTTP(S) requests. More details about
+  authentication can be found in [Authentication](#authentication)
+  section. Default value - not set.
+- `TABPY_TRANSFER_PROTOCOL` - transfer protocol. Default value - `http`. If
+  set to `https` two additional parameters have to be specified:
+  `TABPY_CERTIFICATE_FILE` and `TABPY_KEY_FILE`. More details for how to
+  configure TabPy for HTTPS are at [Configuring HTTP vs HTTPS]
+  (#configuring-http-vs-https) section.
+- `TABPY_CERTIFICATE_FILE` to certificate file to run TabPy with. Only used
+  with `TABPY_TRANSFER_PROTOCOL` set to `https`. Default value - not set.
+- `TABPY_KEY_FILE` to private key file to run TabPy with. Only used
+  with `TABPY_TRANSFER_PROTOCOL` set to `https`. Default value - not set.
+- `TABPY_LOG_DETAILS` - when set to `true` additional call information
+  (caller IP, URL, client info, etc.) is logged. Default value - `false`.
+- `TABPY_EVALUATE_TIMEOUT` - script evaluation timeout in seconds. Default
+  value - `30`.
+
+### Configuration File Example
+
+**Note:** _Always use absolute paths for the configuration paths
+settings._
+
+```ini
+[TabPy]
+# TABPY_QUERY_OBJECT_PATH = /tmp/query_objects
+# TABPY_PORT = 9004
+# TABPY_STATE_PATH = <package-path>/tabpy/tabpy_server
+
+# Where static pages live
+# TABPY_STATIC_PATH = <package-path>/tabpy/tabpy_server/static
+
+# For how to configure TabPy authentication read
+# docs/server-config.md.
+# TABPY_PWD_FILE = /path/to/password/file.txt
+
+# To set up secure TabPy uncomment and modify the following lines.
+# Note only PEM-encoded x509 certificates are supported.
+# TABPY_TRANSFER_PROTOCOL = https
+# TABPY_CERTIFICATE_FILE = path/to/certificate/file.crt
+# TABPY_KEY_FILE = path/to/key/file.key
+
+# Log additional request details including caller IP, full URL, client
+# end user info if provided.
+# TABPY_LOG_DETAILS = true
+
+# Configure how long a custom script provided to the /evaluate method
+# will run before throwing a TimeoutError.
+# The value should be a float representing the timeout time in seconds.
+#TABPY_EVALUATE_TIMEOUT = 30
```

Comment:
Please put a space between # and TABPY_EVALUATE_TIMEOUT it is really bothering me that all the other lines have spaces but this one doesn't